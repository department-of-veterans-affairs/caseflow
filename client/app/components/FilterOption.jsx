import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'glamor';

const listStyling = css({
  paddingBottom: 0,
  margin: 0,
  maxHeight: '345px',
  wordBreak: 'break-word',
  width: '218px',
  overflowY: 'auto',
  listStyleType: 'none',
  paddingLeft: 0
});
const listItemStyling = css({
  padding: '1px',
  position: 'relative'
});

const FilterOption = ({ options, setSelectedValue,
  dropdownFilterViewListStyle, dropdownFilterViewListItemStyle }) => {

  const handleChange = (event) => {
    setSelectedValue(event.target.value);
  };

  return <ul {...dropdownFilterViewListStyle} {...listStyling}>
    {options.map((option, index) => {
      return <li className="cf-filter-option-row" key={index} {...dropdownFilterViewListItemStyle} {...listItemStyling}>
        <input
          type="checkbox"
          id={`${index}-${option.value}`}
          value={option.value}
          checked={option.checked}
          onChange={handleChange} />
        <label htmlFor={`${index}-${option.value}`}>
          {option.displayText}
        </label>

      </li>;
    })}
  </ul>;
};

FilterOption.propTypes = {
  options: PropTypes.array.isRequired,
  setSelectedValue: PropTypes.func.isRequired,
  dropdownFilterViewListStyle: PropTypes.object,
  dropdownFilterViewListItemStyle: PropTypes.object
};

export default FilterOption;
