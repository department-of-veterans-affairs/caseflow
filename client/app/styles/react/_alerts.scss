
.usa-alert {

  &.fixed {
    position: fixed;
    top: 0;
    margin-top: 0;
    z-index: 999;
    left: 50%;
    transform: translate(-50%, 0);
  }

  // Remove after USWDS V.2.0
  &.usa-alert-slim {
    background-size: 4rem;

    .usa-alert-body {
      padding-left: 3.5rem;
    }

    .usa-alert-text {
      font-size: 2rem;
    }
  }

  .usa-alert-text {
    line-height: 1.5;
  }
}
