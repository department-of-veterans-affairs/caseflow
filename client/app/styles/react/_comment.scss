.comment-control-button-container {
  display: flex;
  width: 100%;
  padding: 3px 0;
  justify-content: space-between;
}

.comment-save-button-container {
  display: inline-block;
  width: 100%;
  margin-top: 5px;
  margin-bottom: 17px;
}

.comment-control-button {
  padding: 0;

  &.horizontal {
    text-align: left;
  }
}

.comment-control-button-divider {
  padding: 0 10px;
}

.comment-container-selected {
  box-shadow: 3px 3px 5px $color-gray-light;
}

.comment-container {
  word-wrap: break-word;
  word-break: break-word;
  background-color: $color-white;
  padding: 15px;
  cursor: pointer;

  &:not(.comment-horizontal-container) {
    margin-bottom: 15px;
    border-radius: 3px;
    border: 1px solid $color-gray-lighter;
  }

  &.comment-horizontal-container {
    padding: 25px 25px 25px 40px;
  }
}

// Max width is enforced so the size of this comment box
// fits with existing textareas in style guide
.comment-size-container {
  max-width: 430px;

  .cf-form-textarea {
    margin-bottom: 20px;
  }
}

.comment-horizontal-container {
  display: inline-block;
  width: 100%;
}

.comment-horizontal-spacing {
  margin-right: 49px;
}

.commentIcon-container {
  position: absolute;
  z-index: 10;
  // These dimensions need to match the comment icon SVG dimensions.
  width: 40px;
  height: 40px;
}

