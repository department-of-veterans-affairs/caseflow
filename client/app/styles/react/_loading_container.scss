.loadingContainer-positioning {
  position: absolute;
  width: 100%;
  height: 100%;
}

.loadingContainer-table {
  width: 100%;
  height: 100%;
  display: table;
}

.loadingContainer-table-cell {
  text-align: center;
  display: table-cell;
  vertical-align: middle;
}

.loadingContainer-content {
  position: relative;
  z-index: 1;

  iframe {
    width: 100%;
    height: 80vh;

    &.form9Viewer {
      height: 500px;
    }
  }
}

.loadingContainer-container {
  position: relative;
  overflow: hidden;
}

.cf-loading-button-text {
  display: inline-block;
  margin-right: 10px;
  line-height: 0;
  position: relative;
}

.cf-loading-message {
  .loading-message-text {
    display: inline-block;
    font-style: italic;
    font-weight: bolder;
  }

}

.cf-loading-button-symbol {
  display: inline-block;
}
