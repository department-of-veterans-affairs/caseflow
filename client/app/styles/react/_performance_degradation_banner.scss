.usa-banner {
  background-color: $color-gold-lightest;
  text-align: center;
}

.usa-banner-inner {
  padding: .8em;
}

.banner-icon {
  display: inline-block;
  vertical-align: text-top;
  // height: 25px;
}

.banner-text {
  padding-left: 6px;
}
