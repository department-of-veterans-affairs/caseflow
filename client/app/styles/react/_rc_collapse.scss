@mixin common() {

  width: 0;
  height: 0;
  font-size: 0;
  line-height: 0;
}

@mixin right($w, $h, $color) {

  border-top: $w solid transparent;
  border-bottom: $w solid transparent;
  border-left: $h solid $color;
}

@mixin bottom($w, $h, $color) {

  border-left: $w solid transparent;
  border-right: $w solid transparent;
  border-top: $h solid $color;
}

// scss-lint:disable IdSelector
#arrow {
  @include common();
  @include right(3px, 4px, $color-gray);
  @include bottom(3px, 4px, $color-gray);
}

.rc-collapse {

  &-anim-active {
    transition: height .2s ease-out;
  }

  & > &-item {
    &:first-child {
      border-top: none;
    }
  }

  & > &-item-disabled > .rc-collapse-header {
    cursor: not-allowed;
    color: $color-gray-light;
    background-color: $color-gray-lighter;
  }

  &-content {
    overflow: hidden;

    &-inactive {
      display: none;
    }
  }

  &-item:last-child {
    > .rc-collapse-content {
      border-radius: 0 0 3px 3px;
    }
  }

  & > &-item-active {
    > .rc-collapse-header {

      .arrow {

        #arrow > {
          @include bottom(3px, 4px, $color-gray);
        }
      }
    }
  }
}

.usa-accordion-content {
  &:not([aria-hidden]) {
    position: relative;
    left: 0;
  }
}
