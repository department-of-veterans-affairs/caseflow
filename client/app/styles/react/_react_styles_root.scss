@import '_comment.scss';
@import '_search_bar.scss';
@import '_performance_degradation_banner.scss';
@import '_rc_collapse.scss';
@import '_loading_container.scss';
@import '_select-dropdown.scss';
@import '_toggle_button.scss';
