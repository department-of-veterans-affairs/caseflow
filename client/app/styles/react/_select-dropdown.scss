$cf-select-bg-color: #fff;
$cf-select-disabled-bg-color: $color-gray-lightest;
$cf-select-disabled-color: #323a45;
$cf-select-border-color: #5b616b;
$cf-select-text-color: #323a45;

.cf-select {
  .cf-select__control {
    background-color: $cf-select-bg-color;
    border: 1px solid $cf-select-border-color;
    border-collapse: separate;
    border-radius: 0;
    border-spacing: 0;
    color: $cf-select-text-color;
    cursor: default;

    &--is-disabled {
      background-color: $cf-select-disabled-bg-color;

      .cf-select__single-value {
        color: $cf-select-disabled-color;
      }

      .cf-select__indicators {
        opacity: 0.35;
      }

      input {
        margin: 0;
      }
    }

    .cf-select__value-container {
      padding-bottom: 0;
      padding-top: 0;

      .cf-select__placeholder {
        color: $cf-select-text-color;
      }

      // Ensure that the "selected" value is aligned correctly
      .cf-select__placeholder,
      .cf-select__single-value {
        position: unset;
        top: unset;
        transform: unset;
      }
    }

    .cf-select__indicators {
      .cf-select__indicator-separator {
        background-color: $cf-select-border-color;
        margin-bottom: 0;
        margin-top: 0;
      }

      .cf-select__indicator {
        svg {
          fill: $cf-select-text-color;
        }
      }
    }
  }

  .cf-select__menu {
    border-radius: 0;

    .cf-select__menu-list {
      .cf-select__option {
        background-color: $color-white;
        color: $color-gray;

        &.cf-select__option--is-focused {
          background-color: $color-gray;
          color: $color-white;
        }
      }
    }
  }

  .usa-input-error & {
    .cf-select--is-disabled {
      input { width: 1px; }
    }
  }
}
