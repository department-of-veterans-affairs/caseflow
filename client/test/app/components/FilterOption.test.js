import React from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { axe } from 'jest-axe';

import FilterOption from 'app/components/FilterOption';

describe('FilterOption', () => {
  const setSelectedValue = jest.fn();

  const props = {
    options: [
      {
        displayText: 'Attorney Legacy Tasks',
        value: 'AttorneyLegacyTask',
        checked: true,
      },
      {
        displayText: 'Establish Claim',
        value: 'EstablishClaim',
        checked: false,
      },
    ],
    setSelectedValue
  };

  it('renders correctly', () => {
    const { container } = render(<FilterOption {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('passes a11y testing', async () => {
    const { container } = render(<FilterOption {...props} />);

    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });

  it('correctly sets checkbox status when all are unchecked', () => {
    const uncheckedOptions = props.options.map((opt) => {
      return { ...opt, checked: false };
    });
    const allFalseOptionsProps = { ...props, options: uncheckedOptions };

    const component = render(<FilterOption {...allFalseOptionsProps} />);
    const options = component.getAllByRole('checkbox');
    const checked = options.filter((el) => el.checked);

    expect(checked.length).toBe(0);
  });

  it('correctly sets checkbox status when all are checked', () => {
    const checkedOptions = props.options.map((opt) => {
      return { ...opt, checked: true };
    });
    const allTrueOptionsProps = { ...props, options: checkedOptions };

    const component = render(<FilterOption {...allTrueOptionsProps} />);
    const options = component.getAllByRole('checkbox');
    const checked = options.filter((el) => el.checked);

    expect(checked.length).toBe(options.length);
  });

  it('correctly calls setSelectedValue', async () => {
    const component = render(<FilterOption {...props} />);

    const options = component.getAllByRole('checkbox');
    const opt = props.options[1];
    const checkedBefore = options.filter((el) => el.checked);

    expect(checkedBefore.length).toBe(1);

    await userEvent.click(screen.getByLabelText(opt.displayText));

    // Calls onChange handler
    expect(setSelectedValue).toHaveBeenCalledTimes(1);

    // While we aren't updating value, handleChange is still getting called
    expect(setSelectedValue).toHaveBeenLastCalledWith(opt.value);

    const checkedAfter = options.filter((el) => el.checked);

    // Value should not have been updated, since we're not doing that
    expect(checkedAfter.length).toBe(1);
  });
});
