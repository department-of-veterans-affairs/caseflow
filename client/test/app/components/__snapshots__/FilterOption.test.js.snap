// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterOption renders correctly 1`] = `
<div>
  <ul
    data-css-rbmpqz=""
  >
    <li
      class="cf-filter-option-row"
      data-css-ewovsq=""
    >
      <input
        checked=""
        id="0-AttorneyLegacyTask"
        type="checkbox"
        value="AttorneyLegacyTask"
      />
      <label
        for="0-AttorneyLegacyTask"
      >
        Attorney Legacy Tasks
      </label>
    </li>
    <li
      class="cf-filter-option-row"
      data-css-ewovsq=""
    >
      <input
        id="1-EstablishClaim"
        type="checkbox"
        value="EstablishClaim"
      />
      <label
        for="1-EstablishClaim"
      >
        Establish Claim
      </label>
    </li>
  </ul>
</div>
`;
